File: obg.Domain/Entities/Employee.cs
using obg.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace obg.Domain.Entities
{
    public class Employee : User
    {
        public Pharmacy Pharmacy { get; set; }

        public Employee() { }
        public Employee(string name, int code, string email, string password, string address, RoleUser role, string registerDate, Pharmacy pharmacy) : base(name, code, email, password, address, role, registerDate)
        {
            Pharmacy = pharmacy;
        }
    }
}

File: ObgApp/src/app/models/employee.ts
import { Pharmacy } from "./pharmacy";
import { RoleUser, User } from "./user";

export class Employee extends User {

    Pharmacy: Pharmacy | null;

    constructor(Name: string | null, Code: number | null, Email: string | null, Password: string | null, Address: string | null, Role: RoleUser | null, RegisterDate: string | null, Pharmacy: Pharmacy | null) {
        super(Name, Code, Email, Password, Address, Role, RegisterDate);
        this.Pharmacy = Pharmacy;
    }
}